config/routes.rb

```diff
@@ -1,6 +1,7 @@
 OneBody::Application.routes.draw do
 
-  root to: redirect('/stream')
+  if Setting.find_by_name('Stream').value then root to: redirect('/stream')
+  else root to: redirect('/search') end
```

Comment:
I was getting an error with it anyway.  Moving the logic to controllers/ streams_controller